android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/shopping/GeckoShoppingProductAnalysisResult.kt

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.shopping
+
+import mozilla.components.concept.engine.shopping.ShoppingProductAnalysisResult
+
+/**
+ * Holds the result of the analysis of a shopping product.
+ *
+ * @property productId Product identifier (ASIN/SKU)
+ * @property analysisURL Analysis URL
+ * @property grade Reliability grade for the product's reviews
+ * @property adjustedRating Product rating adjusted to exclude untrusted reviews
+ * @property needsAnalysis Boolean indicating if the analysis is stale
+ * @property lastAnalysisTime Time since the last analysis was performed
+ * @property deletedProductReported Boolean indicating if reported that this product has been deleted
+ * @property deletedProduct Boolean indicating if this product is now deleted
+ * @property highlights Object containing highlights for product
+ */
+data class GeckoShoppingProductAnalysisResult(
+    override val productId: String,
+    val analysisURL: String,
+    val grade: String,
```

Comment:
to do: this probably needs to be an enum

Reply:
update: no, it's better not, as we are not sure if the API is not going to change